module/VuFind/src/VuFind/Config/SearchSpecsReader.php

```diff
@@ -37,119 +36,16 @@
-        if (!isset($this->searchSpecs[$filename])) {
-            $this->searchSpecs[$filename] = $this->getFromPaths(
-                Locator::getBaseConfigPath($filename),
-                Locator::getLocalConfigPath($filename)
-            );
-        }
-
-        return $this->searchSpecs[$filename];
-    }
-
-    /**
-     * Given core and local filenames, retrieve the searchspecs data.
-     *
-     * @param string $defaultFile Full path to file containing default YAML
-     * @param string $customFile  Full path to file containing local customizations
-     * (may be null if no local file exists).
-     *
-     * @return array
-     */
-    protected function getFromPaths($defaultFile, $customFile = null)
-    {
-        // Connect to searchspecs cache:
-        $cache = (null !== $this->cacheManager)
-            ? $this->cacheManager->getCache('searchspecs') : false;
-
-        // Generate cache key:
-        $cacheKey = basename($defaultFile) . '-'
-            . (file_exists($defaultFile) ? filemtime($defaultFile) : 0);
-        if (!empty($customFile)) {
-            $cacheKey .= '-local-' . filemtime($customFile);
-        }
-        $cacheKey = md5($cacheKey);
-
-        // Generate data if not found in cache:
-        if ($cache === false || !($results = $cache->getItem($cacheKey))) {
-            $results = $this->parseYaml($customFile, $defaultFile);
-            if ($cache !== false) {
-                $cache->setItem($cacheKey, $results);
-            }
-        }
-
-        return $results;
-    }
-
-    /**
-     * Process a YAML file (and its parent, if necessary).
-     *
-     * @param string $file          YAML file to load (will evaluate to empty array
-     * if file does not exist).
-     * @param string $defaultParent Parent YAML file from which $file should
-     * inherit (unless overridden by a specific directive in $file). None by
-     * default.
-     *
-     * @return array
-     */
-    protected function parseYaml($file, $defaultParent = null)
-    {
-        // First load current file:
-        $results = (!empty($file) && file_exists($file))
-            ? Yaml::parse(file_get_contents($file)) : [];
-
-        // Override default parent with explicitly-defined parent, if present:
-        if (isset($results['@parent_yaml'])) {
-            $defaultParent = $results['@parent_yaml'];
-            // Swallow the directive after processing it:
-            unset($results['@parent_yaml']);
-        }
-
-        // Now load in missing sections from parent, if applicable:
-        if (null !== $defaultParent) {
-            foreach ($this->parseYaml($defaultParent) as $section => $contents) {
-                if (!isset($results[$section])) {
-                    $results[$section] = $contents;
-                }
-            }
-        }
-
-        return $results;
+        parent::__construct($cacheManager);
+        $this->cacheName = 'searchspecs';
```

Comment:
Is there actually a compelling need for the searchspecs to be cached separately from other YAML? Might it make sense to just rename the searchspecs cache to YAML and be done with it? The only advantage of separation I can think of is that it would allow different caches to be cleared independently, but that seems like a very minimal benefit.

It could still potentially make sense to have a SearchSpecsReader subclass of YamlReader since I could imagine we might eventually want to differentiate some other behavior (such as parent inheritance processing) but for now we might be able to get away with making this an empty class.

If we do go down that road, we should also ensure that any special handling of the 'searchspecs' cache (for example, in the Admin module's cache clearing functionality or in the cache manager itself) gets appropriately updated. I'm not sure if such special handling even exists, but it's worth checking for.